File: perl/src/Creational/Singleton/TestSingleSpoon.pl
package Creational::Singleton;


//TestSingleSpoon.java - testing the singleton

class TestSingleSpoon {
    
   public static void main(String[] args) {
       System.out.println("First person getting the spoon");
       SingleSpoon spoonForFirstPerson = 
         SingleSpoon.getTheSpoon();
       if (spoonForFirstPerson != null)
           System.out.println(spoonForFirstPerson);
       else
           System.out.println("No spoon was available");
       
       System.out.println("");
       
       System.out.println("Second person getting a spoon");
       SingleSpoon spoonForSecondPerson = SingleSpoon.getTheSpoon();
       if (spoonForSecondPerson != null)
           System.out.println(spoonForSecondPerson);
       else
           System.out.println("No spoon was available");
       
       System.out.println("");
       
       System.out.println("First person returning the spoon");
       spoonForFirstPerson.returnTheSpoon();       
       spoonForFirstPerson = null;
       System.out.println("The spoon was returned");
       
       System.out.println("");
       
       System.out.println("Second person getting a spoon");
       spoonForSecondPerson = SingleSpoon.getTheSpoon();
       if (spoonForSecondPerson != null)
           System.out.println(spoonForSecondPerson);
       else
           System.out.println("No spoon was available");
   }
}      

File: java/t/test/Creational/TestSingleSpoon.java
package test.Creational;

import static org.junit.Assert.assertEquals;

import org.junit.Test;
import org.junit.Ignore;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

import Creational.Singleton.*;


//TestSingleSpoon.java - testing the singleton

class TestSingleSpoon {

	@Test    
   public void thetester() {
       
       SingleSpoon spoonForFirstPerson = SingleSpoon.getTheSpoon();
       org.junit.Assert.assertNotNull("First spoon instance ", spoonForFirstPerson);
       
       SingleSpoon spoonForSecondPerson = SingleSpoon.getTheSpoon();
       org.junit.Assert.assertNull("Second person getting a spoon",spoonForSecondPerson);
       
       spoonForFirstPerson.returnTheSpoon();       
       //spoonForFirstPerson = null;
       
       spoonForSecondPerson = SingleSpoon.getTheSpoon();
       org.junit.Assert.assertNotNull( "Second person getting a spoon again ", spoonForSecondPerson);
   }
}      
